Atomic/atomic.py

```diff
@@ -516,6 +518,55 @@ def container_run_command(self):
         command += self.image
         return command
 
+    def ps(self):
+        all_containers = []
+
+        # Collect the system containers
+        for i in self.get_system_containers():
+            container = i["Id"]
+            inspect_stdout = util.check_output(["runc", "state", container])
+            ret = json.loads(inspect_stdout)
+            status = ret["status"]
+            if not self.args.all and status != "running":
+                continue
+
+            image = i['Image']
+            command = ""
+            created = dateparse(ret['created']).strftime("%F %H:%M") # pylint: disable=no-member
+            all_containers.append({"type" : "systemcontainer", "container" : container,
+                                   "image" : image, "command" : command, "created" : created,
+                                   "status" : status})
+
+        # Collect the docker containers
+        for container in [x["Id"] for x in self.d.containers(all=self.args.all)]:
+            ret = self._inspect_container(name=container)
```

Comment:
Have you given any thought to using the .top() api from docker-py to get this sort of information?  One good thing about it is you can add ps arguments to the query which could make your ps function more powerful.  There is an example in atomic Top -> https://github.com/projectatomic/atomic/blob/master/Atomic/top.py#L171